test/shape_test.cpp

```diff
@@ -185,6 +185,31 @@ TEST_CASE(test_shape_packed)
     EXPECT(not s.broadcasted());
 }
 
+TEST_CASE(test_shape_ndim_static)
```

Comment:
Add a unit-test for constructor of `shape` which takes `type`, `max`,`min` and `opts`.